relay-pii/src/processor.rs

```diff
@@ -101,6 +101,47 @@ impl<'a> Processor for PiiProcessor<'a> {
         self.apply_all_rules(meta, state, None)
     }
 
+    fn process_array<T>(
+        &mut self,
+        value: &mut Array<T>,
+        _meta: &mut Meta,
+        state: &ProcessingState<'_>,
+    ) -> ProcessingResult
+    where
+        T: ProcessValue,
+    {
+        // If the array has length 2, we treat it as key-value pair and try to scrub it. If the
+        // scrubbing doesn't do anything, we try to scrub values individually.
+        if value.len() == 2 {
+            if let Some(key) = value[0].clone().into_value() {
+                if let Value::String(key_name) = key.into_value() {
+                    if let Some(inner_value) = value[1].value() {
+                        // We compute a new state which has the first element of the array as the
+                        // key.
+                        let entered = state.enter_borrowed(
+                            key_name.as_str(),
+                            state.inner_attrs(),
+                            inner_value.value_type(),
+                        );
+
+                        let previous_meta = value[1].meta().clone();
+                        process_value(&mut value[1], self, &entered)?;
+                        // We check whether meta has changed from empty to non-empty in order to
+                        // understand if some rules matched downstream.
+                        if previous_meta.is_empty() && !value[1].meta().is_empty() {
```

Comment:
A better impl here would check for structural equivalence but we might gain some performance if we assume that metadata won't be there on an incoming annotated value. Maybe this assumption is totally wrong.